lib/spoom/cli/srb/metrics.rb

```diff
@@ -0,0 +1,68 @@
+# typed: true
+# frozen_string_literal: true
+
+module Spoom
+  module Cli
+    module Srb
+      class Metrics < Thor
+        include Helper
+
+        default_task :show
+
+        desc "show", "Show metrics about Sorbet usage"
+        option :dump, type: :boolean, default: false
+        def show(*paths)
+          files = collect_files(paths)
+          metrics = Spoom::Sorbet::Metrics.collect_code_metrics(files)
+
+          if options[:dump]
+            metrics.counters.sort_by { |key, _value| key }.each do |key, value|
+              puts "#{key} #{value}"
+            end
+
+            return
+          end
+
+          say("Files: `#{files.size}`")
+
+          ["classes", "modules", "singleton_classes"].each do |key|
+            value = metrics[key]
+            next if value == 0
+
+            say("\n#{key.capitalize}: `#{value}`")
+            ["with_srb_type_params", "with_rbs_type_params"].each do |subkey|
+              say(" * #{subkey.gsub("_", " ")}: `#{metrics["#{key}_#{subkey}"]}`")
+            end
+          end
+
+          ["methods", "accessors"].each do |key|
+            value = metrics[key]
+            next if value == 0
+
+            say("\n#{key.capitalize}: `#{value}`")
+            ["without_sig", "with_srb_sig", "with_rbs_sig"].each do |subkey|
+              say(" * #{subkey.gsub("_", " ")}: `#{metrics["#{key}_#{subkey}"]}`")
+            end
+          end
```

Comment:
This feels like we're trying to represent nested data with a flat structure. Would it make sense to have nested hashes for the metrics and then replace all of this code with just a pretty print of the hash?

Reply:
I'm not sure.

The grouping only happens because I want to display the metrics in a cute way.

For most of other uses I have (CodeDB or pasting into a spreadsheet) it's not that important.

I like the simplicity, versatility and extensibility of the flat counters.